.github/workflows/tf-deploy-basic.yaml

```diff
@@ -98,11 +98,20 @@ jobs:
         id: init
         run: |
           tf_args=""
-          if [ "${{ inputs.tf_state_key }}" != "" ] && [ "${{ inputs.tf_state_bucket }}" != "" ]; then
-             tf_args='-backend-config=key=${{ inputs.tf_state_key }} -backend-config=bucket=${{ inputs.tf_state_bucket }}'
+          if [ "${{ inputs.tf_state_bucket }}" != "" ]; then
+            tf_args="$tf_args -backend-config=bucket='${{ inputs.tf_state_bucket }}'"
+          fi
+          if [ "${{ inputs.tf_state_key }}" != "" ]; then
+            tf_args="$tf_args -backend-config=key='${{ inputs.tf_state_key}}'"
```

Comment:
@rbapst-tamedia What if instead we allow for free-form input?

I'm thinking about the use case where we use `-backend-config=backends/dev.s3.tfbackend` with the entire file. Will it be another if condition? I feel like it could easily become ugly.

Reply:
@sgametrio ok. But this was more a fix for my use case. If we use a free-form input, then we have to review the input parameters (tf_init_args instead of "tf_state_bucket/tf_state_key) and change the "client" of this workflow. Do we all agree ?

Reply:
In fact, I will merge this one and let the input change in another PR